packages/commerce-sdk-react/src/auth/storage.ts

```diff
@@ -44,16 +45,28 @@ export class CookieStorage extends BaseStorage {
         super(options)
 
         if (typeof document === 'undefined') {
-            throw new Error('CookieStorage is not avaliable on the current environment.')
+            throw new Error('CookieStorage is not available on the current environment.')
         }
     }
     set(key: string, value: string, options?: Cookies.CookieAttributes) {
         const suffixedKey = this.getSuffixedKey(key)
+        const parentUrl = document.location?.ancestorOrigins?.[0] || document.referrer
```

Comment:
document.location?.ancestorOrigins?.[0] will provide the parent url, but it only works for Chrome and Safari. Firefox does not have this field. `document.referrer` is common use for parent url, but it could be less reliable . It is best to use it a fallback option for Firefox

Reply:
To use `document.referer` we also need to set referrerPolicy for iframe in runtime admin. See [here](https://git.soma.salesforce.com/cc-mobify/runtime-admin-ui/pull/362/files#diff-774c9af1b1ee4cb9420ade1547c0c2a16a59be5ce94db89bc198f678252fb1a2R961)

Reply:
could you add comment to explain why `ancestorOrigins` and why falling back to referrer? this is important knowledge to keep in the code base.